stately/src/otherNativeMain/kotlin/co/touchlab/stately/concurrency/Lock.kt

```diff
@@ -1,9 +1,21 @@
 package co.touchlab.stately.concurrency
 
+import co.touchlab.stately.freeze
 import kotlin.native.concurrent.AtomicInt
+import kotlin.native.concurrent.ThreadLocal
 
+@ThreadLocal
+private object Thread {
+  val id = Any().freeze().hashCode()
```

Comment:
```suggestion
  val id: Any by lazy { Any() }
```
Then just use `===` to compare

Reply:
oh nvm. I see this needs to be an init.

Reply:
I need this to be an Int since i'm using it with an AtomicInt.

Reply:
Okay updated the suggestion. It doesn't need to be frozen, since it's thread-local